### PR TITLE
owner: add qicz to maintainers and move youngnick to emeritus

### LINKED ISSUE #1759: Add @qicz to maintainers
*Description*:

After @qicz becomes eg reviewer, he continues to actively contribute feature/refactor/fixes and code reviews to eg. Also he is engaging to put EG in production and we recieved many valuable feedbacks from real production environment.

So I want to sponsor his maintainer membership, and hopes him can continuely contribute to eg and makes more values in new role.

PRs from him: https://github.com/envoyproxy/gateway/commits?author=qicz

cc @envoyproxy/gateway-maintainers

### PR DESCRIPTION
**Pull request by Xunzhuo**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1759

### CODE REVIEW

**Comment by Xunzhuo** [comment]
cc @envoyproxy/gateway-maintainers 

And also ping @mattklein123 to add @qicz to maintainer team, thanks : )

File: docs/v0.4.0/dev/CODEOWNERS.md [hunk 1]
```diff
@@ -6,10 +6,11 @@
 - @arkodg
 - @skriss
 - @Xunzhuo
-- @youngnick
```
Comment (zirain) [inline docs/v0.4.0/dev/CODEOWNERS.md:9]: we should not change old version doc?

Comment (arkodg) [reply docs/v0.4.0/dev/CODEOWNERS.md:9]: +1

Comment (Xunzhuo) [reply docs/v0.4.0/dev/CODEOWNERS.md]: Make sense, updated/

**Review by zirain — APPROVED** [review]

**Review by arkodg — APPROVED** [review]

**Pull request merged**